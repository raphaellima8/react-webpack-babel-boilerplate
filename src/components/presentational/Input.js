import React, { Fragment } from 'react';
import styled, { css } from 'styled-components';
import InputMask from 'react-input-mask';

const StyledInput = styled.input.attrs(({ type }) => type)`
  ${({ removeDefaultStyle }) =>
    removeDefaultStyle
      ? css`
          width: 100%;
          border: 0;
          padding: 0.5rem 0.5rem 0.5rem 0;
          outline: none;
        `
      : css``};
`;

const ZipCodeFieldContainer = styled.div`
  width: 100%;
  display: flex;
  input {
    width: 80%;
    border: none;
    outline: none;
    color: gray;
  }
  i {
    margin: 1rem;
  }
`;

const Input = ({
  type,
  placeholder,
  term,
  onInputChange,
  removeDefaultStyle
}) => (
  <StyledInput
    type={type}
    value={term}
    onChange={onInputChange}
    placeholder={placeholder}
    removeDefaultStyle={removeDefaultStyle}
  />
);

export const ZipCodeField = ({
  fieldValue,
  onChangeCb,
  placeholder,
  onBlurCb
}) => (
  <ZipCodeFieldContainer>
    <i className="fa fa-truck" aria-hidden="true" />
    <InputMask
      mask="99999-999"
      maskChar={null}
      value={fieldValue}
      onChange={onChangeCb}
      placeholder={placeholder}
      type="search"
      onBlur={onBlurCb}
    />
  </ZipCodeFieldContainer>
);

export default Input;
