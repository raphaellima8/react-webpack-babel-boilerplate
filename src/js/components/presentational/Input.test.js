import React from 'react';
import { shallow } from 'enzyme';

import Input from './Input.js';

test('Test Input Component', () => {
  const component = shallow(
    <Input 
      text="Example text"
      label="example_text"
      type="text"
      id="example_text"
      value="value test"
      handleChange={ () => 'ok' }
    />
  );
  
  expect(component.text()).toEqual('Example text');
  expect(component.type()).toEqual('div');
  
  expect(component).toMatchSnapshot();
});