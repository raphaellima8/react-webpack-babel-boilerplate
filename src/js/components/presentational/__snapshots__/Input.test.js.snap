// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Input Component 1`] = `ShallowWrapper {}`;
